### ISSUE TITLE
Now way to perform correct shutdown

### ISSUE THREAD

**Issue by 13h3r** [akka/reactive-kafka#135]
There is no way to perform correct `Consumer` shutdown in the current API. 

Pseudocode to illustrate:

```scala
val control = Consumer.commitableSource(config)
  .mapAsync(10)(longAsyncTaskWhichTakes1SecondOrSo)
  .mapAsync(batchCommit) //commit once per second
  .to(Sink.ignore)
  .run()
```

The code above starts stream processing. Now there is only `control.stop()` to shut down the `Consumer`. This method waits for no commits in progress and shut down kafka consumer regardless messages processed by my async task.

With this semantic there is no way to perform correct shutdown even when everything is ok.

One of solution can be introducing `stop(finiteDuration)` method, which completes the `Source` immediately, but keeps kafka consumer online for given duration.

**Comment by patriknw** [comment]
ok, I'm not sure that timeout is needed.

We can rename existing `stop()` to `shutdown()` (aligned with `ActorMaterializer.shutdown`) and use `stop()` for this new "stop reading from Kafka" feature. Then you can call `stop()` followed by `shutdown()` and you control the time between those calls.

WDYT?

**Comment by kciesielski** [comment]
@patriknw Do I understand correctly that this means using `Control` in following manner:
````scala
val f = control.stop()
Await.result(f, myCustomDuration)
control.shutdown() // Unit
````
Or maybe we should return `Futures` from both `stop()` and `shutdown()`? By the way, I'm curious why `ActorSystem` has an async `terminate()`, when `ActorMaterializer` only has a `shutdown(): Unit`?

**Comment by patriknw** [comment]
yes, we should return futures from these things
you can use `akka.pattern.after` or ordinary scheduler instead of `Await`

**Comment by 13h3r** [comment]
So lets rename `stop` -> `shutdown` and introduce `stop` with new semantic.

```scala 
trait Control {
  def stop(): Unit
  def shutdown(): Futute[Done]
}
```

I can not figure out why do we need both `stop` and `stopped`. Do we need to keep `stopped`? //cc @drewhk

**Comment by patriknw** [comment]
`stopped` (to be renamed to `isShutdown`) corresponds to a getter for the state, note that it can be shutdown for other reasons, e.g. cancelation from downstreams

**Comment by patriknw** [comment]
can you fix this @13h3r, thanks

**Comment by varadshere** [comment]
How to stop or shutdown Kafka consumer? I tried control.stop()  control.shutdown(), still consumer runs.

**Comment by politrons** [comment]
Yep same situation here, the resolution of the future does not determine that we stop consuming. Only after some time it's really stop the consumption

**Comment by ennru** [comment]
Are you aware of the `akka.kafka.consumer.stop-timeout` setting? 
The actual stopping of the internal actor is deferred so that committing can finish.
